validations/src/validations.rs

```diff
@@ -150,30 +155,162 @@ pub fn validate_rad_request(rad_request: &RADRequest) -> Result<(), failure::Err
     }
 
     let aggregate = &rad_request.aggregate;
-    unpack_radon_script(aggregate.script.as_slice())?;
+    let filters = aggregate.filters.as_slice();
+    let reducer = aggregate.reducer as u8;
+    create_radon_script(filters, reducer)?;
 
     let consensus = &rad_request.tally;
-    unpack_radon_script(consensus.script.as_slice())?;
+    let filters = consensus.filters.as_slice();
+    let reducer = consensus.reducer as u8;
+    create_radon_script(filters, reducer)?;
 
     Ok(())
 }
 
+// Auxiliar counter
+#[derive(Default)]
+struct Counter {
+    max_pos: usize,
+    max_val: i32,
+    values: [i32; 7],
```

Comment:
I don't like the thought of having to change this 7 when we add/remove radon types...

Maybe you can make this a vector and add a constructor like:

```rust
impl Counter {
    fn new(n: usize) {
        let values = vec![0; n];
```

Reply:
And then you can initialize it to `RadonTypes::num_variants()` or something